src/Storefront/Resources/app/storefront/src/plugin/wishlist/wishlist-cookie-offcanvas.plugin.js

```diff
@@ -0,0 +1,167 @@
+            }
+        }, 'left');
+    }
+
+    /**
+     * @private
+     */
+    _registerEvents() {
+        const acceptBtn = this.el.querySelector(this.options.acceptBtnSelector);
+        if (acceptBtn) {
+            acceptBtn.addEventListener('click', this._onAccept.bind(this));
+        }
+
+        const loginBtn = this.el.querySelector(this.options.loginBtnSelector);
+        if (loginBtn) {
+            loginBtn.addEventListener('click', this._onLogin.bind(this));
+        }
+
+        const prefBtn = this.el.querySelector(this.options.prefBtnSelector);
+        if (prefBtn) {
+            prefBtn.addEventListener('click', this._onPreferences.bind(this));
+        }
+
+        const cancelBtn = this.el.querySelector(this.options.cancelBtnSelector);
+        if (cancelBtn) {
+            cancelBtn.addEventListener('click', this._onCancel.bind(this));
+        }
+    }
+
+    /**
+     * @private
+     */
+    _onAccept() {
+        CookieStorageHelper.setItem(
+            this.options.cookieName,
+            '1',
+            this.options.cookieLifetime
+        );
+        this._closeOffcanvas();
+        this.$emitter.publish('WishlistCookie/onAccept', { productId: this.options.productId });
+    }
+
+    /**
+     * @private
+     */
+    _onLogin() {
+        this._closeOffcanvas();
+        window.location.href = window.router['frontend.account.login.page'];
+        this.$emitter.publish('Wishlist/onLoginRedirect');
+    }
+
+    /**
+     * @private
+     */
+    _onPreferences(event) {
+        event.preventDefault();
+        this._closeOffcanvas();
+
+        const configurator = window.PluginManager.getPluginInstances('CookieConfiguration')[0];
+        if (!configurator) {
+            return;
+        }
+
+        const handler = updated => {
+            if (updated[this.options.cookieName]) {
+                this.$emitter.publish('WishlistCookie/onAccept');
+                document.$emitter.unsubscribe('CookieConfiguration_Update', handler);
+            }
+        };
+        document.$emitter.subscribe('CookieConfiguration_Update', handler);
+
+        configurator.openOffCanvas(() => {
+            const offcanvasElement = document.querySelector('.offcanvas');
+            if (!offcanvasElement) {
+                return;
+            }
+
+            offcanvasElement.addEventListener('hidden.bs.offcanvas', () => {
+                document.$emitter.unsubscribe('CookieConfiguration_Update', handler);
+
+                this._restoreFocus();
+            }, { once: true });
+        });
+    }
+
+    /**
+     * @private
+     */
+    _onCancel() {
+        this._closeOffcanvas();
+    }
+
+    _closeOffcanvas() {
+        AjaxOffCanvas.close();
+
+        // remove all offcanvas-backdrop nodes
+        document.querySelectorAll('.offcanvas-backdrop').forEach(el => el.remove());
+    }
+
+    _restoreFocus() {
+        const btn = WishlistCookieOffcanvasPlugin.lastTriggerElement;
+        if (btn && btn.focus) {

```

Comment:
```suggestion
        if (wishlistButton?.focus) {
```
I'm always a fan of readable variable names, so please rather do that. Also `?.` should be a better practise here as well 🙂